code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -453,6 +453,11 @@
 	close_all() //How are you going to see whats inside this thing while throwing it
 	..()
 
+/obj/item/weapon/storage/recycle(var/datum/materials/rec)
+	if(contents)
+		mass_remove(get_turf(src))

```

Comment:
Don't think so.

The issue is that `mass_remove()` is overriden on gadget bags to only drop the lower tier parts.